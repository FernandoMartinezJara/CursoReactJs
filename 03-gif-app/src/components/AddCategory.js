import React, { useState} from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({setcategories}) => {

    const [inputValue, setinputValue] = useState('')

    const handleInputChange = (e) =>{
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim()){
            setcategories(cats => [inputValue, ...cats]);
            setinputValue('')
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input
                value={inputValue}
                onChange={ handleInputChange }
                type="text"/>
        </form>
    )
}

AddCategory.propTypes = {
    setcategories: PropTypes.func.isRequired
}
