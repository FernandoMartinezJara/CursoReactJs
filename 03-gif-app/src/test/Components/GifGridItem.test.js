import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas para componente <GifGridItem />', () => {

    const title = 'Titulo';
    const url = 'https:www.google.cl';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    
    test('debe de mostrar <GifGridItem/> ', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el titulo ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    
    test('debe de obtener una imagen igual al url y all de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe( url );
        expect(img.prop('alt')).toBe( title );
    });
    
    test('debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toBe(true);
    })
    
})
