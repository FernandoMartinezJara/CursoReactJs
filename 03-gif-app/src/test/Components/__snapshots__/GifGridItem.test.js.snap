// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para componente <GifGridItem /> debe de mostrar <GifGridItem/>  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https:www.google.cl"
  />
  <p>
    Titulo
  </p>
</div>
`;
